### DISCUSSION TITLE
Hexes

### DISCUSSION THREAD

**Discussion by markdav-is** [markdav-is/Hexstor#8] [General]
I'm more focused on the underlying tech, so there's lots of room for creativity.  The other primary tech that needs to be built is the hex-based map for the encounters.  Joel mentioned finding that interesting and I shared this link with the raw math for such things: https://www.redblobgames.com/grids/hexagons/.  I was playing a game called "Into the Breach" that has great strat-game elements and used a square grid. it would be great to have hexes in either a top-down or isometric (morphic?) view depending on the graphics skills we have available.

**Comment by markdav-is** [comment]
This SO post has some interesting CSS for dealing with rows and columns of hex and using css for all the settings.  https://stackoverflow.com/questions/10062887/generate-repeating-hexagonal-pattern-with-css3

This could work nicely with "CSS Interop" that dynamically change the content of the hexes for terrain.  all thought I think the ships will want o be free-floating for animation reasons.

**Reply by Vexmage** [reply]
Ah, that’s interesting! A week or so ago, I was experimenting with generating a staggered hex grid just using C# and Blazor. My initial approach handled most of the hex grid calculation in a HexMath.cs file and Home.razor for rendering, but incorporating a CSS Interop could better separate layout from logic, eh?

We could use CSS for the hex layout and terrain styling, while C# dynamically applies classes or content to each hex (marking terrain types, highlighting movement ranges, or handling interactivity with the free-floating ships). 
This separation of concerns seems smart. Thoughts?

**Comment by Vexmage** [comment]
Great, some study and work to dive into! :)
